nni/compression/pytorch/quantization/ptq_quantizer.py

```diff
@@ -0,0 +1,382 @@
+                vmin, vmax = data[layer.name]['input_output'][2], data[layer.name]['input_output'][3]
+                scale, zero_point = calculate_qparams(vmin, vmax,
+                                                      module.output_qmin,
+                                                      module.output_qmax,
+                                                      module.layer_quant_setting.output.quant_scheme,
+                                                      module.layer_quant_setting.output.quant_dtype)
+                module.output_scale.copy_(scale)
+                module.output_zero_point.copy_(zero_point)
+                quant_result_conf[layer.name]['output'] = {'qmin': module.output_qmin, 'qmax': module.output_qmax,
+                                                           'scale': scale, 'zero_point': zero_point}
+        self.compressed = True
+        # for removing hooks
+        self.evaluator.unbind_model()
+        return self.bound_model, quant_result_conf
+
+    def _oneshot_fold_bn(self):
+        for wrapper in self.get_modules_wrapper():
+            if wrapper.bn_module is not None:
+                module = wrapper.module
+                bn_module = wrapper.bn_module
+                weight = module.weight
+                assert hasattr(module, 'bias')
+                bias = module.bias
+                new_weight, new_bias = self._fold_bn(bn_module, weight, bias)
+                module.weight.copy_(new_weight)
+                module.bias.copy_(new_bias)
+                module.register_buffer('old_bn_weight', module.weight.data)
+                module.register_buffer('old_bn_bias', module.bias.data)
+
+    def fold_bn(self, *inputs, wrapper):
+        """
+        As bn folding has been done in ``_oneshot_fold_bn``, directly return the module's
+        weight and bias
+
+        Parameters
+        ----------
+        inputs : tuple of torch.Tensor
+            inputs for the module
+        wrapper : QuantizerModuleWrapper
+            the wrapper for origin module
+
+        Returns
+        -------
+        Tuple of torch.Tensor
+        """
+        assert self.bound_model is not None
+        module = wrapper.module
+        return module.weight, module.bias
+
+    def _quantize(self, x, scale, zero_point, qmin, qmax):
+        x = x / scale + zero_point
+        x = torch.clamp(x, qmin, qmax)
+        x = torch.round(x)
+        x = (x - zero_point) * scale
+        return x
+
+    def quantize_input(self, inputs, wrapper, **kwargs):
+        if self.compressed:
+            module = wrapper.module
+            inputs = self._quantize(inputs,
+                                    module.input_scale,
+                                    module.input_zero_point,
+                                    module.input_qmin,
+                                    module.input_qmax)
+        return inputs
+
+    def quantize_weight(self, wrapper, **kwargs):
```

Comment:
after quantizer.compress(), if the model still performs inference process, then in the quantizer wrapper, the module.weight will be replaced by module.old_weight.data. maybe you should implement the quant-dequant logic for weights.